delete swc.py

alishakodibagkar opened the pull request:
- Removed swc.py and test_swc.py
  - Made changes to these 11 files to implement Neuron_trace.py:
       - brainlit/utils/__init__.py
       - brainlit/utils/upload.py
       - brainlit/utils/session.py
       - experiments/ffn/generating_image_masks.ipynb
       - experiments/axon_geometry/scripts/generate_trace_data.py
       - tests/test_fit_spline.py
       - tests/test_tube_seg.py
       - brainlit/archive/upload_skeleton.py
       - docs/notebooks/utils/downloading_benchmarking.ipynb
       - docs/notebooks/visualization/neighborhood_visualization_demo.ipynb
       - docs/notebooks/utils/downloading_brains.ipynb
   - Added “get_sub_neuron_paths” to Neuron_trace.py since “get_sub_neuron” returns a subneuron in graph format
       - Added example of get_sub_neuron_paths in NeuronTrace_tutorial.ipynb
       - Added get_sub_neuron_paths to test_Neuron_trace.py, passed pytest
   - Changed “get_segments” in session.py to return in graphs form and paths form
       - Changed test_session.py for testing get_segments in session.py, since I now return the segments in graphs form and paths form
   - Next step that needs to be made: make sure notebooks are cleaned up (often times I didn't have the specific files/filepaths that people used for notebooks so I made sure the changes I made worked using my own paths for swc files or s3 files. This messed up notebook outputs so I need to go back in and try my best to replicate outputs)
   - Questions: 
       * What is [viz.rst](https://github.com/neurodata/brainlit/blob/develop/docs/reference/viz.rst), should I change it?
       * [loading.ipynb](https://github.com/neurodata/brainlit/blob/6410a3b613dabdaa15f0c149b51c7c512019c08e/docs/notebooks/visualization/loading.ipynb) mainly demonstrates swc.py functions. Should I still include this file? If so ,how should I go about editing it?
       * [Installation_on_win.md](https://github.com/neurodata/brainlit/blob/a117ac52efec2224a4311c9167b33d46d90b3f05/docs/Installation_on_Win7.md) shows “detail of code history”  for installation problem, so I didn’t remove swc.py functions from that text. Is that okay?
       * [tests/archive/test_visualize.py](https://github.com/neurodata/brainlit/blob/0544d6159a663ae26099e5ec2bc0aa80a20623af/tests/archive/test_visualize.py) was fully commented out. I assumed this means I don’t need to change it/it's not needed?

alishakodibagkar commented:
-  Removed swc.py functions from viz.rst
-  Removed NeuronTrace_tutorial and added documentation to Neuron_trace.py instead
-  Fixed docs with links to data not working (added data to brainlit/data for one of the docs)
-  Updated loading.ipynb to demonstrate functions of Neuron_trace.py

tathey1 reviewed (changes requested):
A few notes:
- I believe codecov only went down because you are removing a large file. Regardless, can you please increase coverage of NeuronTrace to 90%? You are almost there
- Is there not going to be a Neuron_Trace tutorial anymore? If so, can you please give reasoning here in this PR why you decided that?
- Remove notebook output, you can do this in jupyter notebook with "clear all output" or on the command line with nbstripout, described [here](https://github.com/neurodata/brainlit/issues/84)
- Please add an entry to [this file](https://github.com/neurodata/brainlit/blob/develop/docs/reference/utils.rst), where you just copy the entry for Neuroglancersession, except adapt it to your file, in order to get your docs rendered on our website.

alishakodibagkar commented:
- Code coverage improved for Neuron_trace.py (covered above lines in test)
- I removed the Neuron_Trace tutorial and instead updated brainlit/docs/notebooks/visualization/loading.ipynb. This file previously demonstrated the use of functions from swc.py to load neuron data. I changed the file to demonstrate the use of functions from Neuron_trace.py
- Used nbstripout to remove notebook output for all notebooks under brainlit/docs/notebooks
- Updated brainlit/docs/reference/utils.rst

alishakodibagkar commented on brainlit/utils/Neuron_trace.py:
```diff
@@ -43,6 +43,17 @@ class NeuronTrace:
+
+    Example
+    ----------
+    >>> swc_path = "../../../../Manual-GT/8-01_test_1-5/8-01_test_1/tree_2.swc"
```
changed this

alishakodibagkar commented on brainlit/utils/Neuron_trace.py:
```diff
@@ -410,6 +532,70 @@ def get_sub_neuron(self, bounding_box, spacing=None, origin=None):
+
+        # if origin isn't specified but spacing is, set origin to np.array([0, 0, 0])
+        if type(spacing) == np.ndarray and origin is None:
+            origin = np.array([0, 0, 0])
```
code coverage no longer an issue

alishakodibagkar commented on brainlit/utils/Neuron_trace.py:
```diff
@@ -410,6 +532,70 @@ def get_sub_neuron(self, bounding_box, spacing=None, origin=None):
+
+        # voxel conversion option
+        if type(spacing) == np.ndarray:
+            df_voxel = self._df_in_voxel(self.df, spacing, origin)
```
code coverage no longer an issue

alishakodibagkar commented on docs/notebooks/utils/downloading_brains.ipynb:
```diff
@@ -2,18 +2,9 @@
+   "outputs": [],
    "source": [
     "from brainlit.utils.session import NeuroglancerSession\n",
     "from brainlit.utils.swc import graph_to_paths\n",
```
no swc import in this file

tathey1 replied:
are you sure? - https://github.com/neurodata/brainlit/blob/delete_swc/docs/notebooks/utils/downloading_brains.ipynb

alishakodibagkar replied:
I forgot to commit this change, sorry about that - i just did

tathey1 commented:
Go ahead and fix those merge conflicts and then things will be ready

tathey1 reviewed (approved):
all comments resolved
